### PR TITLE
les: fix fetcher syncing logic

### PR DESCRIPTION
**Pull request by zsfelfoldi**
This PR fixes a bug in the fetcher syncing logic. When nextRequest decided to sync it always did set f.syncing to true and it was left on true even if requestDistributor.queue failed to queue the syncing request and it did not start at all (and therefore also never finished). Now nextRequest does not change the flag, just returns its suggestion, then syncLoop sets it later if the queueing has succeeded.

Note: there is a version with debug prints included for testing purpose:
https://github.com/zsfelfoldi/go-ethereum/tree/fetcher-sync-fix-debug

### CODE REVIEW

**Comment by holiman** [comment]
Can you make a testcase that repros the problem and verifies the fix? Or is that deviously difficult?

**Comment by zsfelfoldi** [comment]
@holiman unfortunately rather difficult. This bug only happens when a server disconnects right after announcing a far enough head that would trigger a downloader syncing. It is not possible to reproduce it reliably unless we put some extra callbacks into the fetcher code in which case it wouldn't really simulate a real world scenario anyway.
Btw I wanted to do a "fuzzer" test for the light downloader/fetcher logic for a long time but didn't have the time to do it yet. That would be the best solution to catch these kind of tiny corner case bugs.

File: les/fetcher.go [hunk 1]
```diff
@@ -209,6 +212,7 @@ func (f *lightFetcher) syncLoop() {
 			f.checkSyncedHeaders(p)
 			f.syncing = false
 			f.lock.Unlock()
+			f.requestChn <- false
```
Comment (rjl493456442) [inline les/fetcher.go]: Why should we emit a loop request here?

Comment (zsfelfoldi) [reply les/fetcher.go]: While syncing we disregard loop requests initiated by new announcements. When we are finished syncing we might have missed a newer head announcement. I've seen this happen. Syncing finished but the client was one block behind the actual head and did not update until another block came. Initiating an extra check after processing new announcements has been re-enabled should fix this scenario.

**Comment by karalabe** [comment]
I'm going to merge this PR as the faucet seems more stable than before. Though would be really nice to figure out why it still managed to hang once.

**Pull request merged**